### PR TITLE
Simple option parser

### PR DESCRIPTION
**Pull request by graeme-winter**
Define convention for importing input experiments and reflections, same for
output, merge in external phil scopes, demo in cli_demo;

Means output can be useful:

```
Grey-Area dials-demo-scripts :( [master] $ dials.cli_demo input.experiments=~/data/i04_bag_training/th_8_2_0*.cbf.gz
input {
  experiments = "/Users/graeme/data/i04_bag_training/th_8_2_0*.cbf.gz"
  reflections = None
}
output {
  experiments = None
  reflections = None
}
verbose = False
```

Here it shows optionally the files for input before actually loading them to
get the experiment models out. Means we can have phil parameters which control
how we open the models up.

### CODE REVIEW

**Comment by graeme-winter** [comment]
Also 

```
Grey-Area dials-demo-scripts :) [master] $ dials.cli_demo input.experiments=~/data/i04_bag_training/th_8_2_0*.cbf.gz input.experiments=~/data/i04-eiger-small/Therm_6_2_master.h5 
input {
  experiments = "/Users/graeme/data/i04_bag_training/th_8_2_0*.cbf.gz"
  experiments = "/Users/graeme/data/i04-eiger-small/Therm_6_2_master.h5"
  reflections = None
}
output {
  experiments = None
  reflections = None
}

```

This is a rustic implementation of a basic idea, but I think could be useful - contrast with 

```
Grey-Area tmp :( $ dials.import ~/data/i04_bag_training/th_8_2_0*.cbf.gz ~/data/i04-eiger-small/Therm_6_2_master.h5 
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
DIALS 2.dev.1586-g453396723
The following parameters have been modified:

input {
  experiments = <image files>
}

--------------------------------------------------------------------------------
  format: <class 'dxtbx.format.FormatNexusEigerDLS16M.FormatNexusEigerDLS16M'>
  format: <class 'dxtbx.format.FormatCBFMiniPilatusDLS6MSN100.FormatCBFMiniPilatusDLS6MSN100'>
  num images: 1028
  sequences:
    still:    0
    sweep:    2
  num stills: 0
--------------------------------------------------------------------------------
Writing experiments to imported.expt
```

File: util/command_line.py [hunk 1]
```diff
@@ -9,6 +11,204 @@
...
+            "parameters": parameter_files,
+            "images": image_files,
+            "unknown": unknown,
+        }
+
+    def __init__(self, phil=None):
+        self._input_experiments = None
+        self._input_reflections = None
+
+        parser = argparse.ArgumentParser()
```
Comment (benjaminhwilliams) [inline util/command_line.py:110]: Is there a reason why we don't inherit `argparse.ArgumentParser` as a superclass of this `OptionParser`?

**Comment by graeme-winter** [comment]
While this is stale, I wonder if it would still merge?

**Comment by graeme-winter** [comment]
I have now rebased this on main

File: src/dials/util/command_line.py [hunk 2]
```diff
@@ -8,6 +10,205 @@
...
+        experiment_file_endings = ".expt".split()
+        reflection_file_endings = ".refl".split()
+        parameter_file_endings = ".params .phil".split()
+        image_file_endings = ".h5 .nxs .cbf .img".split()
+        image_compression = ["", ".gz", ".bz2"]
+
+        experiment_files = [
+            filename
+            for filename in filenames
+            if any(filename.endswith(exten) for exten in experiment_file_endings)
```
Comment (benjaminhwilliams) [inline src/dials/util/command_line.py:63]:
```suggestion
            if filename.endswith(experiment_file_endings)
```

**Comment by graeme-winter** [comment]
Closing as this is "not the right shape" apparently

**Pull request closed without merging**